aten/src/ATen/native/quantized/cpu/qnnpack/src/deconv-run.cc

```diff
@@ -173,45 +118,32 @@ enum pytorch_qnnp_status qnnpackDeConv(
           output_min,
           output_max);
 
-  deconvolution->ukernel_type = pytorch_qnnp_ukernel_type_conv;
-  deconvolution->format = pytorch_qnnp_format_quint8;
-
   // Setup the kernel
   const std::array<size_t, 2> output_dims =
       deconv_p.compute_output_dims({input_width, input_height});
   const size_t output_width = output_dims[0];
   const size_t output_height = output_dims[1];
   const size_t kernel_size = kernel_height * kernel_width;
   const size_t output_size = output_height * output_width;
-  const size_t groups = deconvolution->groups;
-  const size_t output_tile_size = pytorch_qnnp_params.q8conv.mr;
-  const size_t tiled_output_size = round_up(output_size, output_tile_size);
-  const size_t indirection_buffer_size =
-      sizeof(void*) * batch_size * groups * tiled_output_size * kernel_size;
 
-  deconvolution->batch_size = batch_size;
-  deconvolution->input_height = input_height;
-  deconvolution->input_width = input_width;
-  deconvolution->input = input;
-  deconvolution->input_pixel_stride = deconv_p.input_channels;
-  deconvolution->output_height = output_height;
-  deconvolution->output_width = output_width;
-  deconvolution->output = output;
-  deconvolution->output_pixel_stride = deconv_p.output_channels;
+  const size_t input_pixel_stride = deconv_p.input_channels;
+  const size_t output_pixel_stride = deconv_p.output_channels;
 
-  const void** indirection_buffer = (const void**)realloc(
-      deconvolution->indirection_buffer, indirection_buffer_size);
-  if (indirection_buffer == NULL) {
+  pytorch_qnnp_status status = pytorch_qnnp_setup_deconvolution2d_nhwc_q8(
```

Comment:
So again, all of the logic removed to the left is replaced with `pytorch_qnnp_setup_deconvolution2d_nhwc_q8`?